migrator/migrations/m_169_to_m_170_move_scope_to_collection_in_report_configurations/migration.go

```diff
@@ -0,0 +1,229 @@
+					MatchType: storage.MatchType_EXACT,
+				})
+			}
+			selectorRules = append(selectorRules, &storage.SelectorRule{
+				FieldName: fieldName,
+				Operator:  storage.BooleanOperator_OR,
+				Values:    ruleValues,
+			})
+		}
+		col := buildEmbeddedCollection(scopeName, index, selectorRules)
+		collections = append(collections, col)
+		index = index + 1
+	}
+	return collections, true
+}
+
+func creatCollectionsToEmbedFromScope(scope *storage.SimpleAccessScope) ([]*storage.ResourceCollection, bool) {
+	collectionsToEmbed := make([]*storage.ResourceCollection, 0)
+
+	index := 0
+	if includedClusters := scope.GetRules().GetIncludedClusters(); len(includedClusters) > 0 {
+		ruleVals := make([]*storage.RuleValue, 0, len(includedClusters))
+		for _, cluster := range includedClusters {
+			ruleVals = append(ruleVals, &storage.RuleValue{
+				Value:     cluster,
+				MatchType: storage.MatchType_EXACT,
+			})
+		}
+		col := buildEmbeddedCollection(scope.GetName(), index, []*storage.SelectorRule{
+			{
+				FieldName: search.Cluster.String(),
+				Operator:  storage.BooleanOperator_OR,
+				Values:    ruleVals,
+			},
+		})
+		collectionsToEmbed = append(collectionsToEmbed, col)
+		index = index + 1
+	}
+
+	if includedNamespaces := scope.GetRules().GetIncludedNamespaces(); len(includedNamespaces) > 0 {
+		for _, namespace := range includedNamespaces {
+			col := buildEmbeddedCollection(scope.GetName(), index, []*storage.SelectorRule{
+				{
+					FieldName: search.Cluster.String(),
+					Operator:  storage.BooleanOperator_OR,
+					Values: []*storage.RuleValue{
+						{
+							Value:     namespace.GetClusterName(),
+							MatchType: storage.MatchType_EXACT,
+						},
+					},
+				},
+				{
+					FieldName: search.Namespace.String(),
+					Operator:  storage.BooleanOperator_OR,
+					Values: []*storage.RuleValue{
+						{
+							Value:     namespace.GetNamespaceName(),
+							MatchType: storage.MatchType_EXACT,
+						},
+					},
+				},
+			})
+			collectionsToEmbed = append(collectionsToEmbed, col)
+			index = index + 1
+		}
+	}
+
+	if clusterLabelSelectors := scope.GetRules().GetClusterLabelSelectors(); len(clusterLabelSelectors) > 0 {
+		labelCollections, success := labelSelectorsToCollections(scope.GetName(), index, clusterLabelSelectors, search.ClusterLabel.String())
+		if !success {
+			return nil, false
+		}
+		index = index + len(labelCollections)
+		collectionsToEmbed = append(collectionsToEmbed, labelCollections...)
+	}
+
+	if namespaceLabelSelectors := scope.GetRules().GetNamespaceLabelSelectors(); len(namespaceLabelSelectors) > 0 {
+		labelCollections, success := labelSelectorsToCollections(scope.GetName(), index, namespaceLabelSelectors, search.NamespaceLabel.String())
+		if !success {
+			return nil, false
+		}
+		collectionsToEmbed = append(collectionsToEmbed, labelCollections...)
+	}
+	return collectionsToEmbed, true
+}
+
+func moveScopeIDToCollectionIDInReports(db *pgxpool.Pool) error {
```

Comment:
Will this feature be enabled at 74?
We shall not have conditional feature flag within a migration. If a feature is disabled at an earlier version and then enabled in a later version, we do not have a chance to redo this migration in upgrade, leaving the table in later version with feature on intact.

Reply:
Actually, @c-du can you confirm it. As far as the README goes, we are good with `m_` and what matters is only the DB sequence number.

@c-du, this will not be conditioned on feature flag but rather postgres env var.

Reply:
Received confirmation from David Shrewsberry that `m` migration is the way to go.

Reply:
@c-du @md2119 I have removed the collections feature flag check here. Does it need any postgres env var check now that postgres will not bet enabled by default in 3.74? This migration should only run if postgres is enabled.